library/alloc/src/vec/in_place_collect.rs

```diff
@@ -193,12 +193,16 @@ where
 
         // Drop any remaining values at the tail of the source but prevent drop of the allocation
         // itself once IntoIter goes out of scope.
-        // If the drop panics then we also leak any elements collected into dst_buf.
+        // If the drop panics then we also try to drop the destination buffer and its elements.
+        // This is safe because `forget_allocation_drop_remaining` forgets the allocation *before*
+        // trying to drop the remaining elements.
```

Comment:
This relies on a guarantee that `forget_allocation_drop_remaining` _will not panic_ before having relinquished the allocation. The panic behavior should be added to the method docs.